### ISSUE TITLE
Assimp fails to load this OBJ

### ISSUE THREAD

**Issue by jankrassnigg** [assimp/assimp#2063]
A while ago I used a very old version of assimp and it could load this file just fine. Recently I updated to the latest official release (December 17), and it stopped working. It seems it does not find any meshes in the file. It may not officially "fail" with some error code, but the number of meshes in the imported data is zero.

Today I updated to the latest assimp snapshot from GitHub and tried again, still cannot import this file.

The log output for importing is this:
Info,  T16648: Load C:/Engine/Trunk/Data/TempContent/id/Imp_data/Imp.obj
Info,  T16648: Found a matching importer for this file format: Wavefront Object Importer.
Info,  T16648: Import root directory is 'C:/Engine/Trunk/Data/TempContent/id/Imp_data\'
Error, T16648: Obj: Ignoring empty face
Skipping one or more lines with the same contents
Info,  T16648: Entering post processing pipeline
Info,  T16648: Leaving post processing pipeline

After that it returns a valid aiScene, but the number of meshes is 0 (everything else is zero as well, except for a single root node with identity transform).

[Imp.zip](https://github.com/assimp/assimp/files/2209894/Imp.zip)

**Comment by aavenel** [comment]
Hi,
I've had a look at your file and it seems that it's encoded with UTF16. I think currently assimp only works with UTF8 ?
I'm not sure if it's a bug or just working as intended. There is a test model that have the same behavior (https://github.com/assimp/assimp/blob/master/test/models/OBJ/box_UTF16BE.obj).
You might be able to fix this if you implement a custom IO system.

**Comment by jankrassnigg** [comment]
Oh, you are correct, didn't notice the different encoding. Switching to Utf-8 makes it work.
Maybe assimp could at least print an error message, notifying the user that any other encoding is unsupported?

**Comment by kimkulling** [comment]
Good point. We provided a UTF-16 encoding for Obj, but this was put by  my into comments. I will fix this issue.

Thanks for the report!

**Comment by tellypresence** [comment]
Similar problem: up until a year ago assimp was able to load this .obj file just fine:
https://free3d.com/3d-model/wwi-plane-pfalz-diii-a-90772.html

But at some point in late 2016 or early 2017, assimp seems to have became more strict and 
refuses to load this model, reporting (IIRC) "invalid number of components".

**Comment by kimkulling** [comment]
The problem in caused by the utf-16 encoding of these files. I tried to decode them to utf-8 and everything works fine. I am currently trying to develop a fix for that.

The issue was introduced when I added a streaming class for reading bigger files blockwise. I forgot to implement the utf-8 encoding, when a windows decoding style like utf-16 is used.

BY the way: the windows unicode support sucks!

**Comment by jankrassnigg** [comment]
Lol, tell me about it. Or maybe don't ;-)

I use this for all the Unicode decoding and encoding:
http://utfcpp.sourceforge.net

This way I don't need platform dependent code.

And for reference, this is what I use that for:
https://github.com/ezEngine/ezEngine/blob/master/Code/Engine/Foundation/Strings/UnicodeUtils.h
https://github.com/ezEngine/ezEngine/blob/master/Code/Engine/Foundation/Strings/StringConversion.h

**Comment by tellypresence** [comment]
Just noticed that this [model](https://free3d.com/3d-model/wwi-plane-pfalz-diii-a-90772.html) started working again
<img src="https://user-images.githubusercontent.com/26958898/109372085-f26f6880-78ea-11eb-805f-bafb7c5f2c23.jpg" width=320 alt="Pfalz">

Was the UTF16 problem fixed recently?

_Note: before the breakage this model was a reliable loader; now more often than not get OpenGL OOM crashes when attempting to load this model_

**Comment by Iam-Jane** [comment]
I meet the same problem! And my obj file is UTF-8! There are other reasons!

**Comment by hernot** [comment]
same problem here even when recoding to utf16 mesh is empty no vertices, no triangles, no normals no triangle_uvs no texture loaded. using assimp through [open3d](https://github.com/intel-isl/Open3D)

**Comment by tellypresence** [comment]
Attempting to load [`Imp.obj`](https://github.com/assimp/assimp/files/2209894/Imp.zip) fails with error
```
Validation failed: aiScene::mNumMeshes is 0. At least one mesh must be there
```
Wondering if that particular `.obj` file is incorrect; since the complicated UTF-16 Pfalz model loads OK closing issue
